src/org/sosy_lab/common/configuration/ConfigurationBuilder.java

```diff
@@ -250,35 +253,72 @@ public ConfigurationBuilder loadFromFile(Path file)
    * @throws IllegalArgumentException If the resource cannot be found or read, or contains invalid
    *     syntax or #include directives.
    */
+  @SuppressWarnings("try")
   public ConfigurationBuilder loadFromResource(Class<?> contextClass, String resourceName) {
     URL url = Resources.getResource(contextClass, resourceName);
     CharSource source = Resources.asCharSource(url, StandardCharsets.UTF_8);
 
     setupProperties();
 
     // Get the path to the source, used for error messages and resolving relative path names.
-    @Var Path sourcePath;
-    @Var String sourceString;
     try {
-      sourcePath = Paths.get(url.toURI());
-      sourceString = sourcePath.toString();
-    } catch (URISyntaxException | FileSystemNotFoundException | IllegalArgumentException e) {
+      // try to access the requested file.
+      URI uri = url.toURI();
+      try (FileSystem fs = getFileSystemForUri(uri)) {
```

Comment:
This closes the `FileSystem` afterwards, but the documentation of `FileSystemProvider.newFileSystem` says

> Once a file system is closed it is provider-dependent if the provider allows a new file system to be created with the same URI as a file system it previously created.

So this could mean that calling this method twice fails, right? Or is it guaranteed for the provider for JAR files that it will succeed to reopen the file twice? Is there any documentation on this?

Furthermore, if the file system already exists before this method is called, won't this even mean that we close it although somebody else has registered it?

Reply:
Yes, in general it depends on the FileSystems whether it can be reopened.
For JAR files, a ZipFileSystem is used and it seems to support multiple openings and closings.

Parallel access might be a problem: we could close a first instance and an interleaved concurrent reading process is then aborted. There is still not a solution for that.

I will change the code to not close, when already opened from other callee.